src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs

```diff
@@ -27,6 +31,114 @@ internal abstract class AbstractRecommendationServiceRunner<TSyntaxContext>
+
+            // Check that a => a. belongs to an invocation.
+            // Find its' ordinal in the invocation, e.g. ThenInclude(a => a.Something, a=> a.
+            var lambdaSyntax = containingMethod.DeclaringSyntaxReferences.Single().GetSyntax(_cancellationToken);
+            if (!(syntaxFactsService.IsAnonymousFunction(lambdaSyntax) &&
+                syntaxFactsService.IsArgument(lambdaSyntax.Parent) &&
+                syntaxFactsService.IsInvocationExpression(lambdaSyntax.Parent.Parent.Parent)))
+            {
+                return default;
+            }
+
+            var invocationExpression = lambdaSyntax.Parent.Parent.Parent;
+            var arguments = syntaxFactsService.GetArgumentsOfInvocationExpression(invocationExpression);
+            var ordinalInInvocation = arguments.IndexOf(lambdaSyntax.Parent);
+
+            var invocation = _context.SemanticModel.GetSymbolInfo(invocationExpression, _cancellationToken);
+            var candidateSymbols = invocation.GetAllSymbols();
+
+            // parameter.Ordinal is the ordinal within (a,b,c) => b.
+            // For candidate symbols of (a,b,c) => b., get types of all possible b.
+            var parameterTypeSymbols = GetTypeSymbols(candidateSymbols, ordinalInInvocation: ordinalInInvocation, ordinalInLambda: parameter.Ordinal);
+
+            // For each type of b., return all suitable members.
+            return parameterTypeSymbols
+                .SelectMany(parameterTypeSymbol => GetSymbols(parameterTypeSymbol, position, excludeInstance: false, useBaseReferenceAccessibility: false))
+                .ToImmutableArray();
+        }
+
+        /// <summary>
+        /// Tries to get a type of its' <paramref name="ordinalInLambda"/> lambda parameter of <paramref name="ordinalInInvocation"/> argument for each candidate symbol.
+        /// </summary>
+        /// <param name="candidateSymbols">symbols corresponding to <see cref="Expression{Func}"/> or <see cref="Func{some_args, TResult}"/>
+        /// Here, some_args can be multi-variables lambdas as well, e.g. f((a,b) => a+b, (a,b,c)=>a*b*c.Length)
+        /// </param>
+        /// <param name="ordinalInInvocation">ordinal of the arguments of function: (a,b) or (a,b,c) in the example above</param>
+        /// <param name="ordinalInLambda">ordinal of the lambda parameters, e.g. a, b or c.</param>
+        /// <returns></returns>
+        protected ImmutableArray<ITypeSymbol> GetTypeSymbols(ImmutableArray<ISymbol> candidateSymbols, int ordinalInInvocation, int ordinalInLambda)
+        {
+            var builder = ArrayBuilder<ITypeSymbol>.GetInstance();
+            var expressionSymbol = _context.SemanticModel.Compilation.GetTypeByMetadataName(typeof(Expression<>).FullName);
+            var funcSymbol = _context.SemanticModel.Compilation.GetTypeByMetadataName(typeof(Func<>).FullName);
```

Comment:
we need to be resilient to these not being around.

it's rare, but it can definitely happen during solution load (when we don't have all references properly attached).  Either we can bail here if they're null, or we can check them in the loop accordingly.